test-integration/operator-tests/src/test/java/org/infinispan/operator/MinimalSetupIT.java

```diff
@@ -0,0 +1,193 @@
+package org.infinispan.operator;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.http.entity.ContentType;
+import org.assertj.core.api.Assertions;
+import org.infinispan.Caches;
+import org.infinispan.Infinispan;
+import org.infinispan.Infinispans;
+import org.infinispan.TestServer;
+import org.infinispan.identities.Credentials;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import cz.xtf.client.Http;
+import cz.xtf.core.http.Https;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.core.waiting.SimpleWaiter;
+import cz.xtf.junit5.annotations.CleanBeforeAll;
+import io.fabric8.kubernetes.api.model.Pod;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * MinimalSetupIT
+ *
+ * Tests all the defaults as minimal infinispan cluster deployment was created.
+ * It is expected that endpoints will be unencrypted with default authentication.
+ *
+ * spec:
+ *   expose:
+ *     type: Route
+ *   replicas: 2
+ *
+ * Should result in the same as:
+ *
+ * spec:
+ *   replicas: 2
+ *   container:
+ *     cpu: 500m
+ *     memory: 512Mi
+ *   service:
+ *     type: cache
+ *     replicationFactor: 2
+ */
+@Slf4j
+@CleanBeforeAll
+class MinimalSetupIT {
+   private static final OpenShift openShift = OpenShifts.master();
+   private static final Infinispan infinispan = Infinispans.minimalSetup();
+   private static final TestServer testServer = TestServer.get();
+   private static final String testServerHost = testServer.host();
+
+   private static String appName;
+   private static String hostName;
+
+   private static String user;
+   private static String pass;
+
+   @BeforeAll
+   static void deployInfinispanCluster() throws IOException {
+      appName =  infinispan.getClusterName();
+      hostName = openShift.generateHostname(appName + "-external");
+
+      infinispan.deploy();
+
+      testServer.withSecret(appName + "-cert-secret");
+      testServer.deploy();
+
+      infinispan.waitFor();
+      testServer.waitFor();
+
+      Credentials developer = infinispan.getDefaultCredentials();
+      user = developer.getUsername();
+      pass = developer.getPassword();
+
+      Https.doesUrlReturnOK("http://" + testServerHost + "/ping").waitFor();
+      Https.doesUrlReturnCode("https://" + hostName, 200).waitFor();
+   }
+
+   /**
+    * Ensure that all resource created by Operator in this tests are deleted.
+    */
+   @AfterAll
+   static void undeploy() throws IOException {
+      infinispan.delete();
+
+      BooleanSupplier bs = () -> openShift.apps().statefulSets().withName(appName).get() == null &&
```

Comment:
This looks like very common for each cluster deletion and can be placed in separate class which will be extended by all tests.

Reply:
Right. I wouldn't say inheritance is the proper solution in here as the expected deletions differ slightly from deployment to deployment and I tried to avoid test inheritance to keep the code as clear as possible. But moving it into common utility class and composing it into `@AfterAll` call would definitely help.